vraptor-core/src/main/java/br/com/caelum/vraptor/ioc/cdi/extension/EagerApplicationScopedExtension.java

```diff
@@ -0,0 +1,38 @@
+package br.com.caelum.vraptor.ioc.cdi.extension;
+
+import java.util.Set;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.context.spi.CreationalContext;
+import javax.enterprise.event.Observes;
+import javax.enterprise.inject.spi.AfterDeploymentValidation;
+import javax.enterprise.inject.spi.Bean;
+import javax.enterprise.inject.spi.BeanManager;
+import javax.enterprise.inject.spi.Extension;
+import javax.enterprise.util.AnnotationLiteral;
+
+import br.com.caelum.vraptor.ioc.cdi.EagerInitialized;
+
+/**
+ * When enabled, it allows users to use {@link EagerInitialized} annotation 
+ * to get your application scoped beans initialized just after CDI deployment.
+ *
+ * @author Rodrigo Turini
+ */
+public class EagerApplicationScopedExtension implements Extension {
+	
+	@SuppressWarnings("serial")
+	public void initialize(@Observes AfterDeploymentValidation event, BeanManager beanManager) {
+		
+		Set<Bean<?>> beans = beanManager.getBeans(Object.class, 
+				new AnnotationLiteral<EagerInitialized>() {});
+		
+		for (Bean<?> bean : beans) {
+			Class<?> beanClass = bean.getBeanClass();
+			if(beanClass.isAnnotationPresent(ApplicationScoped.class)){
+				CreationalContext<?> ctx = beanManager.createCreationalContext(bean);
+				beanManager.getReference(bean, beanClass, ctx).toString();
```

Comment:
@asouza I'm not sure if I get your point. The idea is just to reuse `@Startup` annotation? If so, it would be really confusing and frustrating, since we'd add ejb-api and all its annotations/interfaces only to reuse a single annotation. It'll allow users adding `@DependsOn` trying to order its startup beans, or even use `@Stateless` and so on (that will fail quietly, since it's not implemented). It'd also allow users to use wrong annotations on interceptors, just like `javax.ejb.aroundInvoke`, etc. IMO it doesn't worst. 

If I got the wrong idea, could you elaborate?

Sent from my iPhone

> On 03/02/2015, at 23:43, Alberto Luiz Souza <notifications@github.com> wrote:
> 
> In vraptor-core/src/main/java/br/com/caelum/vraptor/ioc/cdi/extension/EagerApplicationScopedExtension.java:
> 
> > + *
> > + * @author Rodrigo Turini
> > + */
> > +public class EagerApplicationScopedExtension implements Extension {
> > +	
> > +	@SuppressWarnings("serial")
> > +	public void initialize(@Observes AfterDeploymentValidation event, BeanManager beanManager) {
> > +		
> > +		Set<Bean<?>> beans = beanManager.getBeans(Object.class, 
> > +				new AnnotationLiteral<EagerInitialized>() {});
> > +		
> > +		for (Bean<?> bean : beans) {
> > +			Class<?> beanClass = bean.getBeanClass();
> > +			if(beanClass.isAnnotationPresent(ApplicationScoped.class)){
> > +				CreationalContext<?> ctx = beanManager.createCreationalContext(bean);
> > +				beanManager.getReference(bean, beanClass, ctx).toString();
> I think we should use the @Startup annotation. It has a well defined meaning... Just add the ejb-spec jar and it is done. I don't think it is a good practice to create annotations that already exists. Em ter, 3 de fev de 2015 20:26, Lucas Cavalcanti <notifications@github.com> escreveu:
> c
> \
> Reply to this email directly or view it on GitHub.
>